libs/wingc/src/lsp/completions.rs

```diff
@@ -102,29 +24,428 @@ pub unsafe extern "C" fn wingc_on_completion(ptr: u32, len: u32) -> u64 {
 		WASM_RETURN_ERROR
 	}
 }
+
 pub fn on_completion(params: lsp_types::CompletionParams) -> CompletionResponse {
-	FILES.with(|files| {
+	CompletionResponse::Array(FILES.with(|files| {
```

Comment:
We need a way to test these LSP functions -- is this something we could add in the PR (or create an issue for)?

For hovering for example, perhaps we could create a test where we have some Wing fixture file (that we can toss as many edge cases and scenarios into), and write a series of unit tests that are each "given X cursor position in the file, here is a snapshot of the completions that would be generated". WDYT?

Reply:
Adding a separate issue https://github.com/winglang/wing/issues/1779

I'm in full agreement of course, but the fixture loading is a whole can of worms that I think deserves its own PR. With the completion PR merged, I think it should be reasonable to pause feature work and focus on LSP testing.